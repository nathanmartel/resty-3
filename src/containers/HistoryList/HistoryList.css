.historyContainer {
  background-color: #eee;
  padding: 1.5rem;
  width: 100%;
  height: 100%;

  p { 
    font-size: 0.7em;
    margin-bottom: 0;
  }

  button {
    font-size: 0.7em;
  }

}