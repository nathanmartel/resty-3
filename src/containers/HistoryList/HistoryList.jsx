import React from 'react';
import { useLSHistory } from '../../hooks/AppProvider/AppProvider';
import { useLoadHistoryItem } from '../../hooks/History/History';
import HistoryItem from '../../components/HistoryItem/HistoryItem';
import styles from './HistoryList.css';

const HistoryContainer = () => {

  const history = useLSHistory();

  const handleClearHistory = () => {
    localStorage.setItem('history', JSON.stringify([]));
  }; 

  const historyObj = history.map((item, index) => 
    <HistoryItem 
      key={index} 
      method={item.method} 
      url={item.url} 
      index={index} 
      onLoadHistoryItemClick={() => useLoadHistoryItem(index)} 
    />);
  
  return (
    <>
      <div className={styles.historyContainer}>
        <h3>History</h3>
        <hr />
        {historyObj}
        {history.length > 0 && 
          <button onClick={() => handleClearHistory()}>Clear History</button> 
        }
      </div>
    </>
  );
};

export default HistoryContainer;
