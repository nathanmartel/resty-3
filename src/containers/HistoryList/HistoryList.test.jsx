import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  it('renders HistoryList', () => {
    const wrapper = shallow(<HistoryList 
      history={[{
        authPassword: '',
        authToken: '',
        authType: '',
        authUsername: '',
        body: '',
        method: 'GET',
        url: 'http://jsonplaceholder.typicode.com/posts/5',
      }]} 
      onLoadHistoryItemClick={() => {}} 
      onClearHistoryClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
