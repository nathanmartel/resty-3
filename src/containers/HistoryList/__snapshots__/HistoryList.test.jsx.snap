// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<Fragment>
  <div
    className="historyContainer"
  >
    <h3>
      History
    </h3>
    <hr />
    <HistoryItem
      index={0}
      key="0"
      method="GET"
      onLoadHistoryItemClick={[Function]}
      url="http://jsonplaceholder.typicode.com/posts/5"
    />
    <button
      onClick={[Function]}
    >
      Clear History
    </button>
  </div>
</Fragment>
`;
